packages/webapp/pages/[userId]/index.tsx

```diff
@@ -1,306 +1,104 @@
+import request from 'graphql-request';
+import { graphqlUrl } from '@dailydotdev/shared/src/lib/config';
 import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
-import Rank from '@dailydotdev/shared/src/components/Rank';
-import { RANKS, RankHistoryProps } from '@dailydotdev/shared/src/lib/rank';
-import CommentsSection from '@dailydotdev/shared/src/components/profile/CommentsSection';
-import PostsSection from '@dailydotdev/shared/src/components/profile/PostsSection';
-import AuthorStats from '@dailydotdev/shared/src/components/profile/AuthorStats';
-import { Dropdown } from '@dailydotdev/shared/src/components/fields/Dropdown';
-import { useViewSize, ViewSize } from '@dailydotdev/shared/src/hooks';
-import { weeklyGoal } from '@dailydotdev/shared/src/lib/constants';
-import { ButtonSize } from '@dailydotdev/shared/src/components/buttons/ButtonV2';
-import CalendarHeatmap from '../../components/CalendarHeatmap';
+import { RanksWidget } from '@dailydotdev/shared/src/components/profile/RanksWidget';
+import { useActivityTimeFilter } from '@dailydotdev/shared/src/hooks/profile/useActivityTimeFilter';
+import { ReadingTagsWidget } from '@dailydotdev/shared/src/components/profile/ReadingTagsWidget';
+import { ReadingHeatmapWidget } from '@dailydotdev/shared/src/components/profile/ReadingHeatmapWidget';
+import {
+  generateQueryKey,
+  RequestKey,
+} from '@dailydotdev/shared/src/lib/query';
+import { getProfile } from '@dailydotdev/shared/src/lib/user';
 import {
   getLayout as getProfileLayout,
-  getStaticProps as getProfileStaticProps,
   getStaticPaths as getProfileStaticPaths,
+  getStaticProps as getProfileStaticProps,
   ProfileLayoutProps,
-} from '../../components/layouts/ProfileLayout';
-
-export const getStaticProps = getProfileStaticProps;
-export const getStaticPaths = getProfileStaticPaths;
-
-const readHistoryToValue = (value: UserReadHistory): number => value.reads;
-const readHistoryToTooltip = (
-  value: UserReadHistory,
-  date: Date,
-): ReactNode => {
-  const formattedDate = date.toLocaleString('en-US', {
-    month: 'short',
-    day: '2-digit',
-    year: 'numeric',
-  });
-  if (!value?.reads) {
-    return `No posts read on ${formattedDate}`;
-  }
-  return (
-    <>
-      <strong>
-        {value.reads} article{value.reads > 1 ? 's' : ''} read
-      </strong>
-      &nbsp;on {formattedDate}
-    </>
+} from '../../components/layouts/ProfileLayout/v2';
+import { Readme } from '../../../shared/src/components/profile/Readme';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const ProfilePage = ({
+  user: initialUser,
+}: ProfileLayoutProps): ReactElement => {
+  const { user: loggedUser, tokenRefreshed } = useContext(AuthContext);
+  const isSameUser = loggedUser?.id === initialUser?.id;
+
+  const {
+    selectedHistoryYear,
+    setSelectedHistoryYear,
+    before,
+    after,
+    yearOptions,
+    fullHistory,
+  } = useActivityTimeFilter();
+
+  const { data: user } = useQuery(
```

Comment:
I introduced this query also here for users who visit their own profile to make sure the readme is up to date and not cached. Otherwise, it creates a weird UX where you update the readme and it still shows the cached version